### PR TITLE
Storage: Add storage service to rrmngmnt

### PR DESCRIPTION
**Pull request by RazTamir**
Add abbility to perform storage related operations

### CODE REVIEW

File: rrmngmnt/storage.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        Performs 'umount' on 'mount_point' directory and remove
+        the 'mount_point'
+
+        :param mount_point: Path to directory that should be un mounted
+        :type mount_point: str
+        :return: True if umount operation and mount point removal
+        succeeded, False otherwise
+        :rtype: bool
+        """
+        rc, out, err = self.host.executor().run_cmd(
```
Comment (myakove) [inline rrmngmnt/storage.py]: I think self.host have run_command, please use it

File: rrmngmnt/storage.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            ['umount', mount_point, '-f', '-v']
+        )
+        if rc:
+            self.logger.error(
+                "failed to umount directory: %s, output: %s, error: %s",
+                mount_point, out, err
+            )
+            return False
+        rc = self.host.fs.rmdir(mount_point)
```
Comment (myakove) [inline rrmngmnt/storage.py]: what if I don't want to remove the directory, remember that this resource is not related to our tests anymore. I think that it is better to get this as parameter

File: rrmngmnt/storage.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        :param file_name: The file to create
+        :type file_name: str
+        :param path: The path that the file will create under
+        :type path: str
+        :returns: True if succeeded in creating file requested,
+        False otherwise
+        :rtype: bool
+        """
+        command = CREATE_FILE_CMD % (path, file_name)
+        return self.host.executor().run_cmd(shlex.split(command))
```
Comment (myakove) [inline rrmngmnt/storage.py]: did you wanted to return the rc here and not the tuple?

File: rrmngmnt/storage.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        __author__ = "ratamir"
+        :param file_name: The file to create
+        :type file_name: str
+        :param path: The path that the file will create under
+        :type path: str
+        :returns: True if succeeded in creating file requested,
+        False otherwise
+        :rtype: bool
+        """
+        command = CREATE_FILE_CMD % (path, file_name)
```
Comment (myakove) [inline rrmngmnt/storage.py]: logger.info

File: rrmngmnt/storage.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        :param target: Path to target directory, if it is omitted then
+        temporary folder is created and it is the responsible of the
+        caller to remove it at the end
+        :type target
+        :param opts: List of options like: ['-t', 'nfs', '-o', 'defaults']
+        :return: Path to mount point if succeeded, None otherwise
+        """
+        if target is None:
+            target = '/tmp/mnt_point'
+        cmd = ['mkdir', target]
```
Comment (myakove) [inline rrmngmnt/storage.py]: logger.info

File: rrmngmnt/storage.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        self.host.executor().run_cmd(cmd)
```
Comment (myakove) [inline rrmngmnt/storage.py]: if mkdir fail?

File: rrmngmnt/storage.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+        cmd = ['mount', source, target]
+        if opts:
+            cmd.extend(opts)
```
Comment (myakove) [inline rrmngmnt/storage.py]: logger.info

File: rrmngmnt/storage.py (hunk 1 above)
Comment (myakove) [inline rrmngmnt/storage.py]: logger.info

File: rrmngmnt/storage.py (hunk 5 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 6 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 7 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 1 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 2 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 3 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Thanks - Done

File: rrmngmnt/storage.py (hunk 4 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

Comment (RazTamir) [inline rrmngmnt/storage.py]: I added it to run_command in Host

File: rrmngmnt/storage.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        :return: Path to mount point if succeeded, None otherwise
+        """
+        target = '/tmp/mnt_point' if target is None else target
+        cmd = ['mkdir', target]
+        self.logger.info(
+            "Creating directory %s to use as a mount point", target
+        )
+        rc, out, err = self.host.run_command(cmd)
+        if rc:
+            self.logger.error(
```
Comment (myakove) [inline rrmngmnt/storage.py]: when using run_command there is no need to logger.error :) just if rc: return False/None/.....

File: rrmngmnt/storage.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+            shlex.split(PVSCAN_CACHE_CMD)
+        )
+        if rc:
+            logger.error(
+                "Error while executing the '%s' command, output is '%s', "
+                "error is '%s'", PVSCAN_CACHE_CMD, output, error
+            )
+            return False
+
+        if output == '':
```
Comment (myakove) [inline rrmngmnt/storage.py]: This is not looking good, maybe try to find a better why to get all volumes.
when using run_command there is no need for logger.info (you added it to the function)

File: rrmngmnt/storage.py (hunk 8 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: You think the logger: "Failed to create directory as a mount point for...." should be outside this function?

Comment (myakove) [inline rrmngmnt/storage.py]: yes, check run_command function, you have there logger.info (by you) and logger error if rc

Comment (RazTamir) [inline rrmngmnt/storage.py]: ok, so I'll move all detailed loggers outside of rrmngmnt

File: rrmngmnt/storage.py [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+                self.logger.error("failed to remove directory %s", mount_point)
+                return False
+        return True
+
+
+class LVMService(Service):
+    """
+    Storage management class to maintain LVM services
+    """
+
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: No need for the extra blank line

File: rrmngmnt/filesystem.py [hunk 11]
```diff
@@ -30,3 +31,24 @@ def rmdir(self, path):
...
+        )[1].split()
+
+    def touch(self, file_name, path):
+        """
+        Creates a file on host
+
+        __author__ = "ratamir"
+        :param file_name: The file to create
+        :type file_name: str
+        :param path: The path that the file will create under
```
Comment (glazarovich) [inline rrmngmnt/filesystem.py]: The path under which the file will be created

File: rrmngmnt/filesystem.py [hunk 12]
```diff
@@ -30,3 +31,24 @@ def rmdir(self, path):
...
+        :type path: str
+        :returns: True if succeeded in creating file requested,
```
Comment (glazarovich) [inline rrmngmnt/filesystem.py]: True when file creation succeeds, False otherwise

File: rrmngmnt/storage.py (hunk 10 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/filesystem.py (hunk 11 above)
Comment (RazTamir) [inline rrmngmnt/filesystem.py]: Done

File: rrmngmnt/filesystem.py (hunk 12 above)
Comment (RazTamir) [inline rrmngmnt/filesystem.py]: Done

File: rrmngmnt/storage.py [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+        temporary folder is created and it is the responsible of the
+        caller to remove it at the end
+        :type target
+        :param opts: List of options like: ['-t', 'nfs', '-o', 'vers=3']
+        :type opts: list
+        :return: Path to mount point if succeeded, None otherwise
+        :rtype: str
+        """
+        target = '/tmp/mnt_point' if target is None else target
+        cmd = ['mkdir', target]
```
Comment (lukas-bednar) [inline rrmngmnt/storage.py]: mkdir  -p target

File: rrmngmnt/storage.py [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+                self.logger.error("failed to remove directory %s", mount_point)
+                return False
+        return True
+
+
+class LVMService(Service):
+    """
+    Storage management class to maintain LVM services
+    """
+    def lv_change(self, vg_name, lv_name, activate=True):
```
Comment (lukas-bednar) [inline rrmngmnt/storage.py]: Please align names of methods, you have lv_change  and pvscan. In my opinion that underscore is redundant.

File: rrmngmnt/storage.py [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+    """
+
+    def mount(self, source, target=None, opts=None):
+        """
+        Mounts source to target mount point
+
+        __author__ = "ratamir"
+        :param source: Full path to source
+        :type source: str
+        :param target: Path to target directory, if it is omitted then
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: if it is omitted then temporary folder is created and it is the responsible of the caller to remove it at the end ->
if omitted, a temporary folder is created instead

File: rrmngmnt/storage.py [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+        temporary folder is created and it is the responsible of the
+        caller to remove it at the end
+        :type target
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: This line is incomplete - :type target: str

File: rrmngmnt/storage.py [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+        :param opts: List of options like: ['-t', 'nfs', '-o', 'vers=3']
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: List of mount options such as ['-t', 'nfs', '-o', 'vers=3']

File: rrmngmnt/storage.py [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+        :type opts: list
+        :return: Path to mount point if succeeded, None otherwise
+        :rtype: str
+        """
+        target = '/tmp/mnt_point' if target is None else target
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: You may want to generate a more random mount point under /tmp in case this function is called multiple times

File: rrmngmnt/storage.py [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+        cmd = ['mkdir', target]
+        self.logger.info(
+            "Creating directory %s to use as a mount point", target
+        )
+        rc, out, err = self.host.run_command(cmd)
+        if rc:
+            self.logger.error(
+                "Failed to create directory as a mount point for. "
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: The sentence is incomplete: Failed to create a directory to be used as a mount point for %s (then pass source, or alternatively the target you tried to create with the appropriate text change)

File: rrmngmnt/storage.py [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+            self.logger.error(
+                "Failed to mount source %s to target %s. Output: %s",
+                source, target, out
+            )
+            return None
+        return target
+
+    def umount(self, mount_point, force=True, remove_mount_point=True):
+        """
+        Performs 'umount' on 'mount_point' directory and remove
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: Performs an 'unmount' on input 'mount_point' directory, and optionally removes 'mount_point'

File: rrmngmnt/storage.py [hunk 21]
```diff
@@ -0,0 +1,127 @@
...
+        the 'mount_point'
+
+        __author__ = "ratamir"
+        :param mount_point: Path to directory that should be un mounted
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: un mounted -> unmounted

File: rrmngmnt/storage.py [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+        :type mount_point: str
+        :param force: True if force an unmount is required (in case of an
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: True if the mount point should be forcefully removed (such as in the case of an unreachable NFS server)

File: rrmngmnt/storage.py [hunk 23]
```diff
@@ -0,0 +1,127 @@
...
+        rc, out, err = self.host.run_command(cmd)
+        if rc:
+            self.logger.error(
+                "Failed to mount source %s to target %s. Output: %s",
+                source, target, out
+            )
+            return None
+        return target
+
+    def umount(self, mount_point, force=True, remove_mount_point=True):
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: force_unmount would be clearer

File: rrmngmnt/storage.py [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+        Performs 'umount' on 'mount_point' directory and remove
+        the 'mount_point'
+
+        __author__ = "ratamir"
+        :param mount_point: Path to directory that should be un mounted
+        :type mount_point: str
+        :param force: True if force an unmount is required (in case of an
+        unreachable NFS system)
+        :type force: bool
+        :param remove_mount_point: True if mount point should be deleted
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: after 'unmount' operation completes

File: rrmngmnt/storage.py [hunk 25]
```diff
@@ -0,0 +1,127 @@
...
+        return True
+
+
+class LVMService(Service):
+    """
+    Storage management class to maintain LVM services
+    """
+    def lv_change(self, vg_name, lv_name, activate=True):
+        """
+        Set the LV attribute 'active' (active or inactive)
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: Activate or deactivate a Logical volume (by setting it's 'active' attribute)

File: rrmngmnt/storage.py [hunk 26]
```diff
@@ -0,0 +1,127 @@
...
+
+        __author__ = "ratamir"
+        :param vg_name: The name of the Volume group under which LV is
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: LV is contained -> the LV resides

File: rrmngmnt/storage.py [hunk 27]
```diff
@@ -0,0 +1,127 @@
...
+        contained
+        :type vg_name: str
+        :param lv_name: The name of the logical volume which needs to be
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: The name of the logical volume which will be activated or deactivated

File: rrmngmnt/storage.py [hunk 28]
```diff
@@ -0,0 +1,127 @@
...
+        activated or deactivated
+        :type lv_name: str
+        :param activate: True in case the lv change command should activate
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: True when the logical volume should be activated, False when it should be deactivated

File: rrmngmnt/storage.py [hunk 29]
```diff
@@ -0,0 +1,127 @@
...
+        the ls, False in case of deactivate it
+        :type activate: bool
+        :returns: True if succeeded, False otherwise
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: True if setting the logical volume 'active' flag succeeded, False otherwise

File: rrmngmnt/storage.py [hunk 30]
```diff
@@ -0,0 +1,127 @@
...
+        :rtype: bool
+        """
+        active = 'y' if activate else 'n'
+        return self.host.run_command(
+            shlex.split(LV_CHANGE_CMD % active, vg_name, lv_name)
+        )[0] == 0
+
+    def pvscan(self):
+        """
+        Execute 'pvscan' in order to get the latest list of volumes
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: the latest list of volumes -> the current list of physical volumes

File: rrmngmnt/storage.py [hunk 31]
```diff
@@ -0,0 +1,127 @@
...
+
+        __author__ = "ratamir"
+        :returns: True if succeeded, False otherwise
```
Comment (glazarovich) [inline rrmngmnt/storage.py]: True if the pvscan command succeded, False otherwise

File: rrmngmnt/storage.py (hunk 15 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 16 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 17 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 13 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 18 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: The user can send a mount point name so all good

File: rrmngmnt/storage.py (hunk 19 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 23 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: This should be like the real option when you call umount --force

File: rrmngmnt/storage.py (hunk 20 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 21 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 22 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 24 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 14 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 25 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 26 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 27 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 28 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 29 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 30 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

File: rrmngmnt/storage.py (hunk 31 above)
Comment (RazTamir) [inline rrmngmnt/storage.py]: Done

**Comment by RazTamir** [comment]
Is is possible that I will add those tests in other patch? because I want to run our plans with the new code on jenkins

**Comment by lukas-bednar** [comment]
@RazTamir it usually works in different order ... please fulfill your promise.

**Comment by RazTamir** [comment]
(: sure I will
